src/idl_gen_cpp.cpp

```diff
@@ -233,6 +233,39 @@ class CppGenerator : public BaseGenerator {
     return keywords_.find(name) == keywords_.end() ? name : name + "_";
   }
 
+  std::string Name(const FieldDef &field) const {
+    // the union type field suffix is immutable
+    static size_t union_suffix_len = strlen(UnionTypeFieldSuffix());
+    // early return if no case transformation required
+    switch (opts_.cpp_object_api_field_case) {
+      default: return EscapeKeyword(field.name);
+      case IDLOptions::Case_Upper:
+      case IDLOptions::Case_Lower: break;
+    }
+    std::string name = field.name;
+    switch (field.value.type.base_type) {
```

Comment:
again, use `if` ? these switches are hard to read because they signal that something more complicated than an `if` is going on.

Reply:
resolved with commit just now